treatment-service/services/experiment_service.go

```diff
@@ -128,38 +128,37 @@ func (es *experimentService) filterByLookupOrder(
 	filtered := matches
 	for _, segmenter := range segmenters {
 		orderedValues := filters[segmenter]
-		if len(orderedValues) == 1 {
-			continue
-		}
-
 		currentFilteredList := []*models.ExperimentMatch{}
 		segmenterType := segmenterTypes[segmenter]
 		for _, transformedValue := range orderedValues {
 			if len(currentFilteredList) == 0 {
-				for _, segmenterMatch := range filtered {
-					segmenterMatchedValue := segmenterMatch.SegmenterMatches[segmenter]
+				for _, experiment := range filtered {
```

Comment:
Thanks for renaming this 👍🏻 , `segmenterMatch` really didn't make any sense as I was reviewing these changes.